org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/cleanup/AddDeprecatedAnnotationCleanUp.java

```diff
@@ -0,0 +1,44 @@
+/*******************************************************************************
+ * Copyright (c) 2022 Red Hat Inc. and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Red Hat Inc. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.jdt.ls.core.internal.cleanup;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.jdt.core.manipulation.CleanUpContextCore;
+import org.eclipse.jdt.core.manipulation.ICleanUpFixCore;
+import org.eclipse.jdt.internal.corext.fix.Java50FixCore;
+
+/**
+ * A cleanup that adds the deprecated annotation to classes/fields/methods that
+ * are marked deprecated in the javadoc.
+ *
+ * @author datho7561
+ */
+public class AddDeprecatedAnnotationCleanUp implements INoConfigCleanUp {
+
+	/* (non-Javadoc)
+	 * @see org.eclipse.jdt.ls.core.internal.cleanup.INoConfigCleanUp#getIdentifier()
+	 */
+	@Override
+	public String getIdentifier() {
+		return "addDeprecatedAnnotation";
```

Comment:
Just as a comment. I like this style of naming, mainly because there's a good chance we'll be having users configure this themselves, assuming clients don't have some wizard readily available. So it would be good if they are short and easy to remember. Something like `java.cleanup.addDeprecatedAnnotation` gets too verbose for my liking.

Reply:
In VS Code, we can provide autocompletion for these by listing them in the package.json, but since other clients don't necessarily have this, it might be a good idea to shorten them. I'm just worried about the names not being clear enough, although I guess if we document the clean ups well enough somewhere that is easy to find we could avoid confusion.

Reply:
Let me know what you think of the new names

Reply:
I think the new names make sense.